src/Chainweb/Pact/PactService.hs

```diff
@@ -262,37 +262,54 @@ initPactService' ver cid chainwebLogger bhDb pdb sqlenv reorgLimit act = do
     logger = P.newLogger loggers $ P.LogName ("PactService" <> show cid)
 
 initialPayloadState
-    :: PayloadCas cas
-    => ChainwebVersion
+    :: Logger logger
+    => PayloadCas cas
+    => logger
+    -> ChainwebVersion
     -> ChainId
     -> PactServiceM cas ()
-initialPayloadState Test{} _ = pure ()
-initialPayloadState TimedConsensus{} _ = pure ()
-initialPayloadState PowConsensus{} _ = pure ()
-initialPayloadState v@TimedCPM{} cid =
-    initializeCoinContract v cid $ genesisBlockPayload v cid
-initialPayloadState v@FastTimedCPM{} cid =
-    initializeCoinContract v cid $ genesisBlockPayload v cid
-initialPayloadState v@Development cid =
-    initializeCoinContract v cid $ genesisBlockPayload v cid
-initialPayloadState v@Testnet04 cid =
-    initializeCoinContract v cid $ genesisBlockPayload v cid
-initialPayloadState v@Mainnet01 cid =
-    initializeCoinContract v cid $ genesisBlockPayload v cid
+initialPayloadState _ Test{} _ = pure ()
+initialPayloadState _ TimedConsensus{} _ = pure ()
+initialPayloadState _ PowConsensus{} _ = pure ()
+initialPayloadState logger v@TimedCPM{} cid =
+    initializeCoinContract logger v cid $ genesisBlockPayload v cid
+initialPayloadState logger v@FastTimedCPM{} cid =
+    initializeCoinContract logger v cid $ genesisBlockPayload v cid
+initialPayloadState logger  v@Development cid =
+    initializeCoinContract logger v cid $ genesisBlockPayload v cid
+initialPayloadState logger v@Testnet04 cid =
+    initializeCoinContract logger v cid $ genesisBlockPayload v cid
+initialPayloadState logger v@Mainnet01 cid =
+    initializeCoinContract logger v cid $ genesisBlockPayload v cid
 
 initializeCoinContract
-    :: forall cas. PayloadCas cas
-    => ChainwebVersion
+    :: forall cas logger. (PayloadCas cas, Logger logger)
+    => logger
+    -> ChainwebVersion
     -> ChainId
     -> PayloadWithOutputs
     -> PactServiceM cas ()
-initializeCoinContract v cid pwo = do
+initializeCoinContract logger v cid pwo = do
     cp <- getCheckpointer
     genesisExists <- liftIO $ _cpLookupBlockInCheckpointer cp (0, ghash)
-    unless genesisExists $ do
+    if genesisExists
+      then do
+          pdb <- asks _psPdb
+          bhdb <- asks _psBlockHeaderDb
+          reloadedCache <- liftIO $
+              withTempSQLiteConnection chainwebPragmas $ \sqlenv ->
+                  initPactService' v cid logger bhdb pdb sqlenv defaultReorgLimit $ do
+                      -- it is reasonable to assume genesis doesn't exist here
+                      validateGenesis
+                      gets _psInitCache
+          psInitCache .= reloadedCache
```

Comment:
these two lines setting the cache should be unnecessary if we're doing a full play through, since the state is updated in `applyPactCmd`. Have you dumped the cache without setting it explicitly like this to confirm?

Reply:
Confirmed that it does not because `initPactService'` here is executing with separate state and returning `()`